examples/temporal_graph_learning/input_encoder.py

```diff
@@ -0,0 +1,101 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from graphstorm import model as gsmodel
+from graphstorm.utils import get_rank
+
+from model_utils import get_unique_nfields
+from model_utils import get_trainable_params
+from model_utils import to_per_field_nfeats
+from model_utils import concat_over_fields
+
+import dgl
+
+
+class NodeEncoderInputLayer(gsmodel.embed.GSNodeInputLayer):
+    """The temporal input encoder layer for all nodes in a heterogeneous graph.
+
+    The input layer adds a linear layer on nodes with node features and the linear layer projects the node
+    features to a specified dimension.
+    Then, the input layer adds learnable embeddings as temporal encoding on the projected node features.
+
+
+    Parameters
+    ----------
+    g: DistGraph
+        The distributed graph
+    feat_size : dict of int
+        The original feat sizes of each node type
+    embed_size : int
+        The embedding size
+    dropout : float
+        The dropout parameter
+    """
+    def __init__(
+        self,
+        g,
+        out_size,
+        feat_size,
+        dropout=0.0,
+    ):
+        super(NodeEncoderInputLayer, self).__init__(g)
+        self.activation = nn.LeakyReLU()
+        self.dropout = nn.Dropout(dropout)
+        self.out_size = out_size
+        self.feat_size = feat_size
+
+        self.fields = get_unique_nfields(g.etypes)
```

Comment:
in this example, do you assume that there is only one relation type originally? the code may only work on a temporal homogeneous graph. we need to make it work for temporal heterogeneous graphs as well.

Reply:
This code works for hetergenious graph with multiple relation types.

Reply:
This code was based on Cramer's code, I think I didn't make much changes on the number of relation types.